environment/lib/near2eth-relay.js

```diff
@@ -104,18 +119,33 @@ class Near2EthRelay {
                 const lastFinalBlockHash = headBlock.header.last_final_block;
                 // The finalized block is not immediately available so we wait for it to become available.
                 let lightClientBlock = null;
+                let currentValidators = null;
                 while (!lightClientBlock) {
                     // @ts-ignore
+                    currentValidators = await this.near.connection.sendJsonRpc('validators', [null]).result;
+                    if (!currentValidators) {
+                        await sleep(300);
+                        continue;
+                    }
                     lightClientBlock = await this.near.connection.provider.sendJsonRpc('next_light_client_block', [lastFinalBlockHash]);
                     if (!lightClientBlock) {
                         await sleep(300);
+                        continue;
+                    }
+                    // Because fetch currentValidators and lightClientBlock isn't atomic, it's possible we happen to
+                    // fetch lightClentBlock cross epoch boundary. Fetch another time to ensure that's not the case.
+                    let currentValidatorsNow = await this.near.connection.sendJsonRpc('validators', [null]).result;
```

Comment:
We should not rely on three RPC requests `validators `, `next_light_client_block`, `validators` being sequentially consistent. Weird things, like implementation quirks of our RPC server or connecting to a different node can break it. Instead we should pass specific block height with  `validators`, if possible, AFAIK all our RPC endpoints allow that (except light client) CC @frol to confirm.

Reply:
If don't use `validators`, we'd have to wait one epoch to know a block + the bp of this block here. Would be great we can know current epoch bps.

Reply:
yes you can pass a specific height here.

Reply:
1. If you pass a block height instead of the `null`, you will get a view specifically pinned to the block height you request.
2. Use `.validators` method instead of a low-level `sendJsonRpc`.

Reply:
Thanks. Good to learn 1. `2` seems more straightforward here, will do that